optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/swingui/ConferenceSchedulingPanel.java

```diff
@@ -64,14 +82,23 @@ public ConferenceSchedulingPanel() {
                 throw new IllegalStateException("Failed to show temp file (" + tempFile + ") in LibreOffice or Excel.", e);
             }
         });
-        add(importConferenceButton);
-        add(publishButton);
-        add(button);
-        add(new JLabel("Changes to that file are ignored unless you explicitly save it there and open it here."));
+
+        setLayout(new GridLayout(12, 1));
```

Comment:
GridLayout with 12 rows. That works well in the original's windows size, but try resizing it to something much smaller or bigger. It needs to be "responsive", so handle differenet window sizes.
Use a vertical BoxLayout instead.

Reply:
The problem with BoxLayout is when window size is too large the panels containing the buttons are stretched too much to fit the window (which results in too much space between the buttons). Also using VerticalGlue didn't reduce the distance enough, that's why I used GridLayout, wdyt?

Reply:
Alternatively, what you can do is use a BorderLayout for the other panel and add a 2th panel in there BorderLayout.NORTH, That 2th panel can uses a 2-row GridLayout. That should make the resizing behave as it should.